components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/RequestCountResource.java

```diff
@@ -65,16 +65,16 @@ public boolean invoke(MessageContext synCtx,
         String yearParameter = Utils.getQueryParameter(synCtx, "year");
         String monthParameter = Utils.getQueryParameter(synCtx, "month");
 
-        if (tryParseInt(yearParameter) != null && tryParseInt(monthParameter) != null) {
-            if (!StringUtils.isEmpty(yearParameter) && !StringUtils.isEmpty(monthParameter)) {
-                return takeRequestCountOfTheMonth(axis2MessageContext, Integer.parseInt(yearParameter),
-                                                  Integer.parseInt(monthParameter));
-            }
-        } else if (StringUtils.isEmpty(yearParameter) && StringUtils.isEmpty(monthParameter)) {
+        if (StringUtils.isEmpty(yearParameter) && StringUtils.isEmpty(monthParameter)) {
```

Comment:
I think we'll have to handle all the cases, both empty, either one empty and both provided. Or else, we'll still get the 'NumberFormatException' log if we just define the year without the month, or the month without the year.

Reply:
Shall we run a query param validation to either have both params or none before this and respond back to the client, saying both or none are expected?

Reply:
Fixed.